arwen/elrondapi/elrondei.go

```diff
@@ -627,3 +656,280 @@ func int64finish(context unsafe.Pointer, value int64) {
 	gasToUse := hostContext.GasSchedule().ElrondAPICost.Int64Finish
 	hostContext.UseGas(gasToUse)
 }
+
+//export executeOnSameContext
+func executeOnSameContext(
+	context unsafe.Pointer,
+	gasLimit int64,
+	addressOffset int32,
+	valueOffset int32,
+	functionOffset int32,
+	functionLength int32,
+	numArguments int32,
+	argumentsLengthOffset int32,
+	dataOffset int32,
+) int32 {
+	instCtx := wasmer.IntoInstanceContext(context)
+	ethContext := arwen.GetEthContext(instCtx.Data())
```

Comment:
Why use `GetEthContext()` instead of `GetErdContext()`? They may be equivalent here, but it using `GetEthContext()` suggests we're about to do something with the Ethereum API in `elrondei.go`.